src/node_mime.cc

```diff
@@ -0,0 +1,369 @@
+        break;
+      }
+    }
+    if (saw_invalid_char && first_invalid_char < trim_right) {
+      this->flags_ |= MIME_FLAGS_INVALID_SUBTYPE;
+      return;
+    }
+  }
+  if (subtype.size() == 0) {
+    this->flags_ |= MIME_FLAGS_INVALID_SUBTYPE;
+    return;
+  }
+  std::transform(type.begin(), type.end(), type.begin(), to_ascii_lower);
+  std::transform(subtype.begin(),
+                  subtype.end(),
+                  subtype.begin(),
+                  to_ascii_lower);
+  while (right <= src.size()) {
+    right++;   // ;
+    for (; right < src.size(); right++) {
+      auto c = src[right];
+      if (!is_ascii_whitespace(c)) break;
+    }
+    left = right;
+    saw_invalid_char = false;
+    for (; right < src.size(); right++) {
+      auto c = src[right];
+      if (c == ';'|| c == '=') {
+        break;
+      } else if (!is_http_token(c)) {
+        saw_invalid_char = true;
+      }
+    }
+    std::string parameterName;
+    if (right < src.size()) {
+      if (!saw_invalid_char) {
+        parameterName = src.substr(left, right - left);
+        std::transform(parameterName.begin(),
+                        parameterName.end(),
+                        parameterName.begin(),
+                        to_ascii_lower);
+      }
+      if (src[right] == ';') {
+        continue;
+      }
+      right++;   // =
+      left = right;
+    }
+    bool saw_invalid_value = false;
+    std::string parameterValue;
+    if (right < src.size()) {
+      if (src[right] == '"') {
+        right++;
+        while (true) {
+          left = right;
+          for (; right < src.size(); right++) {
+            auto c = src[right];
+            if (c == '"' || c == '\\') {
+              break;
+            } else if (!is_http_quoted_string_token(c)) {
+              saw_invalid_value = true;
+            }
+          }
+          parameterValue += src.substr(left, right - left);
+          if (right < src.size() && src[right] == '\\') {
+            right++;
+            if (right < src.size()) {
+              parameterValue += src[right];
+              right++;
+              continue;
+            } else {
+              parameterValue += '\\';
+              break;
+            }
+          } else {
+            break;
+          }
+        }
+        for (; right < src.size(); right++) {
+          auto c = src[right];
+          if (c == ';') {
+            break;
+          }
+        }
+      } else {
+        for (; right < src.size(); right++) {
+          auto c = src[right];
+          if (c == ';') {
+            break;
+          } else {
+            auto is_2_byte = (c & (unsigned char)0xE0) == 0xC0;
+            auto is_3_byte = (c & (unsigned char)0xF0) == 0xE0;
+            auto is_4_byte = (c & (unsigned char)0xF8) == 0xF0;
+            bool is_invalid = false;
+            if (is_2_byte) {
+              // need 2 bits of FF
+              if ((c & (unsigned char)0x1C) != 0) {
```

Comment:
style nit: No C-style casts, you can use `static_cast` but you shouldn’t really need any cast here?

(Also: What’s `FF`?)

Reply:
https://mimesniff.spec.whatwg.org/#http-quoted-string-token-code-point code points that are valid go up to `\u00ff` `utf8` encodes this as 2 bits of the character in the first byte.